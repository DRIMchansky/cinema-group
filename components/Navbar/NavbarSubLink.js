// packages
import styled from 'styled-components'
// components
import Link from 'next/link'
// utils
import { colors, breakpoints } from '../../utils/cssConfig'

const NavbarSubLink = ({ branch, name, href }) => {
  return (
    <Link href={href} passHref>
      <StyledNavbarSubLink>
        <StyledBranch>{branch}</StyledBranch>
        <StyledName>{name}</StyledName>
      </StyledNavbarSubLink>
    </Link>
  )
}

const StyledNavbarSubLink = styled.a`
  display: block;
  padding: 0.4rem 0 0.4rem 0.5rem;
  color: ${colors.white};
  transition: background 0.1s ease;
  text-decoration: none;

  &:hover {
    background: ${colors.lighter};
  }

  @media (min-width: ${breakpoints.medium}) {
    border-radius: 4px;
  }
`
const StyledBranch = styled.span`
  display: block;
  color: ${colors.primary};
  font-size: 0.75rem;
`

const StyledName = styled.span`
  display: block;
  font-size: 1.1rem;
`

export default NavbarSubLink
