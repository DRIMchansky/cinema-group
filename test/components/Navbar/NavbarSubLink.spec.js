import React from 'react'
import { mount } from 'enzyme'
import NavbarSubLink from '../../../components/Navbar/NavbarSubLink'

test('NavbarSubLink renders correctly', () => {
  const wrapper = mount(<NavbarSubLink href="/" />)

  expect(wrapper).toMatchSnapshot()
})
