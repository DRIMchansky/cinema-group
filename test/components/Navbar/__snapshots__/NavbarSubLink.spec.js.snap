// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarSubLink renders correctly 1`] = `
.c0 {
  display: block;
  padding: 0.4rem 0 0.4rem 0.5rem;
  color: #fff;
  -webkit-transition: background 0.1s ease;
  transition: background 0.1s ease;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  background: #324082;
}

.c1 {
  display: block;
  color: #eebbc3;
  font-size: 0.75rem;
}

.c2 {
  display: block;
  font-size: 1.1rem;
}

@media (min-width:960px) {
  .c0 {
    border-radius: 4px;
  }
}

<NavbarSubLink
  href="/"
>
  <Link
    href="/"
    passHref={true}
  >
    <NavbarSubLink__StyledNavbarSubLink
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <a
        className="c0"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <NavbarSubLink__StyledBranch>
          <span
            className="c1"
          />
        </NavbarSubLink__StyledBranch>
        <NavbarSubLink__StyledName>
          <span
            className="c2"
          />
        </NavbarSubLink__StyledName>
      </a>
    </NavbarSubLink__StyledNavbarSubLink>
  </Link>
</NavbarSubLink>
`;
